packages/firestore/src/core/firestore_client.ts

```diff
@@ -435,7 +435,9 @@ export class FirestoreClient {
 
   write(mutations: Mutation[]): Promise<void> {
     const deferred = new Deferred<void>();
-    this.asyncQueue.enqueue(() => this.syncEngine.write(mutations, deferred));
+    this.asyncQueue.enqueueAndForget(() =>
+      this.syncEngine.write(mutations, deferred)
```

Comment:
Could syncEngine hide this Deferred?

Reply:
Not quite.  The spec tests need to be able to wait for the write() to complete locally without waiting for the user callback (after the server ack of the write). So there are two separate async things you might want to wait on.